src/blocks/pacman.rs

```diff
@@ -0,0 +1,53 @@
+use std::time::Duration;
+use std::process::Command;
+
+use block::Block;
+use widgets::text::TextWidget;
+use widget::I3BarWidget;
+
+use serde_json::Value;
+use uuid::Uuid;
+
+
+pub struct Pacman {
+    text: TextWidget,
+    id: String,
+    update_interval: Duration,
+}
+
+impl Pacman {
+    pub fn new(config: Value, theme: Value) -> Pacman {
+        {
+            Pacman {
+                id: Uuid::new_v4().simple().to_string(),
+                update_interval: Duration::new(get_u64_default!(config, "interval", 5), 0),
+                text: TextWidget::new(theme.clone()).with_text(""),
+            }
+        }
+        
+    }
+}
+
+impl Block for Pacman
+{
+    fn update(&mut self) -> Option<Duration> {
+        let output = String::from_utf8(
+            Command::new("sh")
+                .arg("-c")
+                .arg("pacman -Sup 2>/dev/null")
```

Comment:
Hmm, that's weird. It doesn't prompt me for sudo like it does for a regular pacman call. I think that if you use the p flag on there it doesn't require root. When I run `pacman -Su` is does tell me  I need to be root but when I do `pacman -Sup` it doesn't complain. I can implement the other way if you prefer that though.